### PR TITLE
AK-29987 Add new fields for conditional egress NAT rules and policies

### PR DESCRIPTION
**Pull request by camjrichards**

### CODE REVIEW

File: alkira/helper.go [hunk 1]
```diff
@@ -210,14 +210,14 @@ func stringInSlice(a string, list []string) bool {
...
 
-//For infoblox if there is a failed POST for infoblox the backend does not clean up the
-//credentials that were created in preparation for creating the infoblox service. This means
-//if you make the same attempt to create an infoblox there will likely already be a credential name
-//that exists. This throws an error. To avoid that this function will be used to add a random suffix
-//of a-zA-z to the end of the credential name. That way each time an attempt and subsequent failure
-//occurs when creating the infoblox there will be no clash with existing credentials. This is only
-//neccesary because the infoblox credentials are not exposed in the UI. Otherwise the user could
-//manage the credentials themselves.
+// For infoblox if there is a failed POST for infoblox the backend does not clean up the
```
Comment (spikewang) [inline alkira/helper.go:213]: I think this is not related to policy changes?

File: alkira/resource_alkira_connector_internet_exit.go [hunk 2]
```diff
@@ -85,6 +85,16 @@ func resourceAlkiraConnectorInternetExit() *schema.Resource {
 				Default:      "DEFAULT",
 				ValidateFunc: validation.StringInSlice([]string{"DEFAULT", "SRC_IP"}, false),
 			},
+			"egress_ips": {
+				Description: "The types of egress IPs to use with the connector. Current options are `ALKIRA_PUBLIC_IP` or `BYOIP`. " +
+					"If `BYOIP` is one of the options provided byoip_id must also be set.",
```
Comment (spikewang) [inline alkira/resource_alkira_connector_internet_exit.go:90]: byoip_id -> `byoipd_id`. Markdown format for code

File: alkira/resource_alkira_policy_nat.go [hunk 3]
```diff
@@ -65,6 +65,12 @@ func resourceAlkiraPolicyNat() *schema.Resource {
 				Elem:        &schema.Schema{Type: schema.TypeInt},
 				Required:    true,
 			},
+			"category": {
```
Comment (spikewang) [inline alkira/resource_alkira_policy_nat.go:68]: Do we have a default value for this?

**Comment by spikewang** [comment]
@camjrichards you will need to run `make doc` to re-generate the documentation.

File: alkira/resource_alkira_connector_internet_exit.go [hunk 4]
```diff
@@ -172,6 +182,7 @@ func generateConnectorInternetRequest(d *schema.ResourceData, m interface{}) (*a
 	client := m.(*alkira.AlkiraClient)
 
 	billingTags := convertTypeListToIntList(d.Get("billing_tag_ids").([]interface{}))
+	egressTypes := convertTypeListToStringList(d.Get("egress_ips").([]interface{}))
```
Comment (spikewang) [inline alkira/resource_alkira_connector_internet_exit.go:185]: You will also need to add this to the `resourceConnectorInternetExitRead` function

File: alkira/resource_alkira_policy_nat.go [hunk 5]
```diff
@@ -166,6 +172,7 @@ func generatePolicyNatRequest(d *schema.ResourceData, m interface{}) (*alkira.Na
 		IncludedGroups: inGroups,
 		ExcludedGroups: exGroups,
 		NatRuleIds:     natRules,
+		Category:       d.Get("category").(string),
```
Comment (spikewang) [inline alkira/resource_alkira_policy_nat.go:175]: This needs to be added to READ function too

File: alkira/resource_alkira_policy_nat_rule.go [hunk 6]
```diff
@@ -238,6 +251,7 @@ func generatePolicyNatRuleRequest(d *schema.ResourceData, m interface{}) (*alkir
 		Name:        d.Get("name").(string),
 		Description: d.Get("description").(string),
 		Enabled:     d.Get("enabled").(bool),
+		Category:    d.Get("category").(string),
```
Comment (spikewang) [inline alkira/resource_alkira_policy_nat_rule.go:254]: Add this to READ function

File: alkira/resource_alkira_policy_nat.go (hunk 3 above)
Comment (camjrichards) [reply alkira/resource_alkira_policy_nat.go:68]: There isn't a default value, but an empty value is treated the same as `DEFAULT`. I've updated the description to make this clear

File: alkira/resource_alkira_policy_nat_rule.go [hunk 7]
```diff
@@ -158,6 +165,13 @@ func resourceAlkiraPolicyNatRule() *schema.Resource {
 							Type:        schema.TypeBool,
 							Optional:    true,
 						},
+						"egress_type": {
+							Description:  "The egress type to use with the match. Options are are `ALKIRA_PUBLIC_IP` or `BYOIP`.",
```
Comment (spikewang) [inline alkira/resource_alkira_policy_nat_rule.go:169]: If Default value is `NONE`, then the possible options are `ALKIRA_PUBLIC_IP`, `BYOIP` and `NONE`, right?

Comment (camjrichards) [reply alkira/resource_alkira_policy_nat_rule.go:169]: No, I copied the field definition structure and missed updating the default value. There is no default value. Will update

**Review by spikewang — APPROVED** [review]
